### DISCUSSION TITLE
How to call fetchEvents on interval time ?

### DISCUSSION THREAD

**Discussion by nunomazer** [saade/filament-fullcalendar#185] [General]
I have a page with fullCalendar that keep opened and it should refresh the events each minute, how can I call the method on a setInterval or wire.poll ?

**Comment by cawecoy** [comment]
You can do it via JS:

```
// refresh fullcalendar every 1000 milliseconds
setInterval(function () {
    window.dispatchEvent( new Event('filament-fullcalendar--refresh') );
}, 1000);
```

**Reply by newtec-foggia** [reply]
Sorry, I'm new to filament, where do I enter the code you indicated?
I modified the **calendar-widget.blade.php** file but it doesn't refresh
<x-filament-widgets::widget>
    <x-filament::section>
        {{-- Widget content --}}
    </x-filament::section>
    
    {{-- Aggiungi lo script per aggiornare il calendario ogni minuto --}}
    <script>
        document.addEventListener('DOMContentLoaded', function() {
            setInterval(function() {
                window.dispatchEvent(new Event('filament-fullcalendar--refresh'));
            }, 60000); // 60000 millisecondi = 1 minuto
        });
    </script>
</x-filament-widgets::widget>

**Reply by cawecoy** [reply]
A quick way is to add it via Render Hook. Example:

AppServiceProvider.php:

```php
use Filament\Support\Facades\FilamentView;
use Filament\View\PanelsRenderHook;
use Illuminate\Contracts\View\View;

// ...

public function boot(): void
{
    FilamentView::registerRenderHook(
        PanelsRenderHook::PAGE_END,
        fn (): View => view('fc-refresh-script')
    );
}
```

resources/views/fc-refresh-script.blade.php:

```blade
{{-- Aggiungi lo script per aggiornare il calendario ogni minuto --}}
<script>
    document.addEventListener('DOMContentLoaded', function() {
        setInterval(function() {
            window.dispatchEvent(new Event('filament-fullcalendar--refresh'));
        }, 60000); // 60000 millisecondi = 1 minuto
    });
</script>
```

Also, you can scope the render hook to load the script only in specific pages: https://filamentphp.com/docs/3.x/support/render-hooks#scoping-render-hooks

**Reply by newtec-foggia** [reply]
perfect, thanks

**Comment by nunomazer** [comment]
Thank you @cawecoy it's perfect working
